src/EditorFeatures/CSharpTest/Completion/CompletionProviders/Snippets/CSharpSnippetCompletionProviderTests.cs

```diff
@@ -0,0 +1,214 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CSharp.Completion.CompletionProviders.Snippets;
+using Microsoft.CodeAnalysis.Test.Utilities;
+using Roslyn.Test.Utilities;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.Editor.CSharp.UnitTests.Completion.CompletionProviders.Snippets
+{
+    public class CSharpSnippetCompletionProviderTests : AbstractCSharpCompletionProviderTests
+    {
+        internal override Type GetCompletionProviderType()
+            => typeof(CSharpSnippetCompletionProvider);
+
+        [WpfFact, Trait(Traits.Feature, Traits.Features.Completion)]
+        public async Task InsertConsoleSnippetInMethodTest()
+        {
+            var markupBeforeCommit =
+@"class Program
+{
+    public void Method()
+    {
+        Wr$$
+    }
+}";
+
+            var expectedCodeAfterCommit =
+@"class Program
+{
+    public void Method()
+    {
+        Console.WriteLine($$);
+    }
+}";
+            await VerifyCustomCommitProviderAsync(markupBeforeCommit, "Write to the Console", expectedCodeAfterCommit);
+        }
+
+        [WpfFact, Trait(Traits.Feature, Traits.Features.Completion)]
+        public async Task InsertAsyncConsoleSnippetTest()
+        {
+            var markupBeforeCommit =
+@"class Program
+{
+    public async Task MethodAsync()
+    {
+        Wr$$
+    }
+}";
+
+            var expectedCodeAfterCommit =
+@"class Program
+{
+    public async Task MethodAsync()
+    {
+        await Console.Out.WriteLineAsync($$);
+    }
+}";
+            await VerifyCustomCommitProviderAsync(markupBeforeCommit, "Write to the Console", expectedCodeAfterCommit);
+        }
+
+        [WpfFact, Trait(Traits.Feature, Traits.Features.Completion)]
+        public async Task InsertConsoleSnippetGlobalTest()
+        {
+            var markupBeforeCommit =
+@"
+$$
+class Program
+{
+    public async Task MethodAsync()
+    {
+    }
+}";
+
+            var expectedCodeAfterCommit =
+@"
+Console.WriteLine($$);
```

Comment:
It's kind of unclear here, is this test for top level statements? Does this work for the following scenario?

```csharp
Console.WriteLine("Top level statement writeline");

class One
{
}

$$ // Does snippet show here?

class Two
{
}
```

Reply:
yes it was a test for top level statements. I can add another test for this scenario as well.